worlds/smw/__init__.py

```diff
@@ -243,8 +283,17 @@ def extend_hint_information(self, hint_data: typing.Dict[int, typing.Dict[int, s
                     if level_index >= world_cutoffs[i]:
                         continue
 
-                    if self.multiworld.dragon_coin_checks[self.player].value == 0 and "Dragon Coins" in loc_name:
+                    if self.options.dragon_coin_checks.value == 0 and "Dragon Coins" in loc_name:
                         continue
+                    if self.options.moon_checks.value == 0 and "3-Up Moon" in loc_name:
```

Comment:
not quite sure if it makes sense in this context but in case it does
```suggestion
                    if not self.options.moon_checks and "3-Up Moon" in loc_name:
```